q2_feature_table/_core_features/_visualizer.py

```diff
@@ -0,0 +1,171 @@
+        raise ValueError('min_fraction (%r) parameter must be less than '
+                         'max_fraction (%r) parameter.' %
+                         (min_fraction, max_fraction))
+
+    index_fp = os.path.join(TEMPLATES, 'index.html')
+    context = {
+        'num_samples': table.shape[1],
+        'num_features': table.shape[0]
+    }
+
+    if min_fraction == max_fraction:
+        fractions = [min_fraction]
+    else:
+        fractions = np.linspace(min_fraction, max_fraction, steps)
+
+    rounded_fractions = _round_fractions(fractions)
+
+    data = []
+    file_links = []
+    for fraction, rounded_fraction in zip(fractions, rounded_fractions):
+        core_features = _get_core_features(table, fraction)
+        core_feature_count = len(core_features)
+        data.append([fraction, core_feature_count])
+
+        if core_feature_count > 0:
+            core_feature_fn = 'core-features-%s.tsv' % rounded_fraction
+            core_feature_fp = os.path.join(output_dir, core_feature_fn)
+
+            file_links.append("<a href='./%s'>TSV</a>" % core_feature_fn)
+
+            core_features.to_csv(core_feature_fp, sep='\t',
+                                 index_label='Feature ID')
+        else:
+            file_links.append('No core features')
+
+    df = pd.DataFrame(data, columns=['Fraction of samples', 'Feature count'])
+    df['Fraction of features'] = df['Feature count'] / table.shape[0]
+    df['Feature list'] = file_links
+
+    ax = sns.regplot(data=df, x='Fraction of samples', y='Feature count',
+                     fit_reg=False)
+
+    # matplotlib will issue a UserWarning if attempting to set left and right
+    # bounds to the same value.
+    if min_fraction != max_fraction:
+        ax.set_xbound(min(fractions), max(fractions))
+    ax.set_ybound(0, max(df['Feature count']) + 1)
+
+    ax.get_figure().savefig(
+        os.path.join(output_dir, 'core-feature-counts.svg'))
+
+    table_html = _df_to_html(df, index=False, escape=False, border=0,
+                             classes=("table table-striped table-hover"))
+    context['table_html'] = table_html
+
+    q2templates.render(index_fp, output_dir, context=context)
+
+
+def _get_core_features(table, fraction):
+    filter_f = _get_filter_to_core_f(table, fraction)
+    feature_filtered_table = table.filter(
+        filter_f, axis='observation', inplace=False)
+    index = []
+    data = []
+    for values, id_, _ in feature_filtered_table.iter(axis='observation'):
+        index.append(id_)
+        data.append(_seven_number_summary(values))
+    if len(data) > 0:
+        return pd.DataFrame(data, index=index).sort_values(
+            by='50%', ascending=False)
+    else:
+        return pd.DataFrame()
+
+
+def _get_filter_to_core_f(table, fraction):
+    if not (0.0 <= fraction <= 1.0):
+        raise ValueError("Invalid fraction passed to core filter: %r is "
+                         "outside of range [0,1]." % fraction)
```

Comment:
> Assuming min_fraction and max_fraction are in [0, 1] could np.linspace ever produce values that would hit this branch?
>
> The range predicates on the QIIME type will enforce the value on the min/max arguments, but if we wanted to check anyways, it seems like this logic should just be at the top of the method.

Good catch! This validation is unnecessary so I'll remove it.

> Anecdotally, the compositional statistics book "Modeling and Analysis of Compositional Data" calls these kinds of values proportions very consistently. (Not important for this PR, but it's the closest I've seen as a name for the [0, 1] range.)

Discussed offline with @ebolyen and @thermokarst. It's kind of a toss-up between using *fraction of X* vs *proportion of X* so we'll keep using *fraction* for consistency with other QIIME 2 plugins (and QIIME 1 scripts). We can revisit this later if there's need to switch terms.